src/builtin.c

```diff
@@ -396,21 +396,30 @@ static jv f_divide(jq_state *jq, jv input, jv a, jv b) {
   }
 }
 
+#define dtoi(n) ((n) < INTMAX_MIN ? INTMAX_MIN : -(n) < INTMAX_MIN ? INTMAX_MAX : (intmax_t)(n))
```

Comment:
Oh, `intmax_t`...  `intmax_t` has to be used very carefully, and... you are using it carefully.  Good!